### PR TITLE
Add lifter and uploads to composer

### PR DESCRIPTION
**Pull request by bcl**
This is @evan-goode PR #826 with the related commits squashed, and a pile of tests added.

### CODE REVIEW

**Comment by bcl** [comment]
Here's some more changes/tests. Not ready for review yet, but @jgiardino and @jkozol should take a look at the start compose json where I added profile or settings. there is also a new /upload/providers/delete/ route so now profiles can be removed.

File: docs/html/_sources/composer-cli.rst.txt [hunk 1]
```diff
@@ -60,3 +64,115 @@ Downloading the final image is done with ``composer-cli compose image UUID`` and
 save the qcow2 image as ``UUID-disk.qcow2`` which you can then use to boot a VM like this::
 
     qemu-kvm --name test-image -m 1024 -hda ./UUID-disk.qcow2
+
+
+Image Uploads
+-------------
+
+``composer-cli`` can upload the images to a number of services, including AWS,
+OpenStack, and VSphere. The upload can be started when the build is finished,
```
Comment (chris1984) [inline docs/html/_sources/composer-cli.rst.txt]: S/VSphere/vSphere

File: docs/composer-cli.rst [hunk 2]
```diff
@@ -60,3 +64,115 @@ Downloading the final image is done with ``composer-cli compose image UUID`` and
 save the qcow2 image as ``UUID-disk.qcow2`` which you can then use to boot a VM like this::
 
     qemu-kvm --name test-image -m 1024 -hda ./UUID-disk.qcow2
+
+
+Image Uploads
+-------------
+
+``composer-cli`` can upload the images to a number of services, including AWS,
+OpenStack, and VSphere. The upload can be started when the build is finished,
```
Comment (chris1984) [inline docs/composer-cli.rst]: s/VSphere/vSphere

File: docs/man/composer-cli.1 [hunk 3]
```diff
@@ -245,6 +281,165 @@ qemu\-kvm \-\-name test\-image \-m 1024 \-hda ./UUID\-disk.qcow2
 .fi
 .UNINDENT
 .UNINDENT
+.SH IMAGE UPLOADS
+.sp
+\fBcomposer\-cli\fP can upload the images to a number of services, including AWS,
+OpenStack, and VSphere. The upload can be started when the build is finished,
```
Comment (chris1984) [inline docs/man/composer-cli.1]: s/VSphere/vSphere

**Comment by chris1984** [comment]
I rekicked jenkins and that is passing now :)

**Pull request merged**